### ISSUE TITLE
JavadocMethod: False positive from comment inside method

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#6273]
Identified at https://github.com/checkstyle/checkstyle/issues/6272 ,

JavadocMethod has a mutator that is surviving at `isCommentNodesRequired`.
http://rveach.no-ip.org/checkstyle/regression/pitest-reports/24/javadoc/com.puppycrawl.tools.checkstyle.checks.javadoc/JavadocMethodCheck.java.html#org.pitest.mutationtest.report.html.SourceFile@29d1b4da_339

When looking into it, yes the check is not comments aware because it access FileContents directly and has direct access to the source code. When removing this check and running regression, I found the below false positive where an AST comment is interfering with proper validation.

````
$ cat TestClass.java
public class TestClass {
    @Deprecated
    /** */
    public void foo14() { }

    @Deprecated
    /** */
    public void foo15() { /** */ }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="JavadocMethod" />
    </module>
</module>

$ java -jar checkstyle-8.15-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:2:5: Missing a Javadoc comment. [JavadocMethod]
Audit done.
Checkstyle ends with 1 errors.
````

The 2 cases are exactly the same and so I am expecting the same result. The only difference between the 2 is the 2nd method has a comment inside it.
This is causing the violation on line 6 to be suppressed. I am expecting violations for both lines as the javadocs are similar.

**Comment by rnveach** [comment]
According to https://stackoverflow.com/a/19585281/1016482 , javadocs that come after annotations are not recognized by the javadoc tool, so in addition to removing comment support for this check, there shouldn't be any violations at all.

**Comment by romani** [comment]
Attention, we are ok to open task on deprecated implementation, but keep in mind that whole code inside need to be redone.

**Comment by romani** [comment]
Fix is merged
